MSAL/src/telemetry/MSALTelemetry.m

```diff
@@ -22,57 +22,40 @@
 // THE SOFTWARE.
 
 #import "MSALTelemetry.h"
-#import "MSIDTelemetryEventInterface.h"
-#import "MSALDefaultDispatcher.h"
-#import "MSIDTelemetry.h"
-#import "MSIDTelemetry+Internal.h"
+#import "MSALGlobalConfig.h"
+#import "MSALTelemetryConfig.h"
 
 @implementation MSALTelemetry
 
-- (id)initInternal
-{
-    return [super init];
-}
-
 + (MSALTelemetry *)sharedInstance
 {
-    static dispatch_once_t once;
-    static MSALTelemetry* singleton = nil;
-    
-    dispatch_once(&once, ^{
-        singleton = [[MSALTelemetry alloc] initInternal];
-    });
-    
-    return singleton;
+    return [[self.class alloc] init];
```

Comment:
Do we want this to be a singleton?

Reply:
Why not. It really shouldn't matter that we all methods call end up calling a class methods. 
Nonetheless though, nothing bad comes about having a singleton here :)